dbms/src/Functions/FunctionsCoding.h

```diff
@@ -1276,6 +1276,100 @@ class FunctionUnhex : public IFunction
+    }
+
+    bool useDefaultImplementationForConstants() const override { return true; }
+
+    void executeImpl(Block & block, const ColumnNumbers & arguments, size_t result, size_t input_rows_count) override
+    {
+        auto col_str = ColumnString::create();
+        ColumnString::Chars & out_vec = col_str->getChars();
+        ColumnString::Offsets & out_offsets = col_str->getOffsets();
+
+        const auto size_per_row = arguments.size() + 1;
+        out_vec.resize(size_per_row * input_rows_count);
+        out_offsets.resize(input_rows_count);
+
+        for (size_t row = 0; row < input_rows_count; ++row)
+        {
+            out_offsets[row] = size_per_row + out_offsets[row - 1];
+            out_vec[row * size_per_row + size_per_row - 1] = '\0';
+        }
+
+        Columns columns_holder(arguments.size());
+        for (size_t idx = 0; idx < arguments.size(); ++idx)
+        {
+            //partial const column 
+            columns_holder[idx] = block.getByPosition(arguments[idx]).column->convertToFullColumnIfConst();
+            const IColumn * column = columns_holder[idx].get();
+
+            if (!(executeNumber<UInt8>(*column, out_vec, idx, input_rows_count, size_per_row)
+                || executeNumber<UInt16>(*column, out_vec, idx, input_rows_count, size_per_row)
+                || executeNumber<UInt32>(*column, out_vec, idx, input_rows_count, size_per_row)
+                || executeNumber<UInt64>(*column, out_vec, idx, input_rows_count, size_per_row)
+                || executeNumber<Int8>(*column, out_vec, idx, input_rows_count, size_per_row)
+                || executeNumber<Int16>(*column, out_vec, idx, input_rows_count, size_per_row)
+                || executeNumber<Int32>(*column, out_vec, idx, input_rows_count, size_per_row)
+                || executeNumber<Int64>(*column, out_vec, idx, input_rows_count, size_per_row)
+                || executeNumber<Float32>(*column, out_vec, idx, input_rows_count, size_per_row)
+                || executeNumber<Float64>(*column, out_vec, idx, input_rows_count, size_per_row)))
+            {
+                throw Exception{"Illegal column " + block.getByPosition(arguments[idx]).column->getName()
+                                + " of first argument of function " + getName(), ErrorCodes::ILLEGAL_COLUMN};
+            }
+        }
+
+        block.getByPosition(result).column = std::move(col_str);
+    }
+
+private:
+    template <typename T>
+    bool executeNumber(const IColumn & src_data, ColumnString::Chars & out_vec, const size_t & column_idx, const size_t & rows, const size_t & size_per_row)
+    {
+        const ColumnVector<T> * src_data_concrete = checkAndGetColumn<ColumnVector<T>>(&src_data);
+
+        if (!src_data_concrete)
+        {
+            return false;
+        }
+
+        for (size_t row = 0; row < rows; ++row)
+        {
+            out_vec[row * size_per_row + column_idx] = static_cast<char>(src_data_concrete->getInt(row));
```

Comment:
Ok. BTW, it's in question what variant of loop is faster - this one or permuted.

Reply:
missed that..